src/gatherers/service-worker.js

```diff
@@ -41,8 +43,18 @@ class ServiceWorker extends Gather {
     });
   }
 
+  static getOrigin(url) {
+    if (typeof window.URL === 'function') {
+      return new window.URL(url).origin;
+    }
+
+    const parsedURL = require('url').parse(url);
```

Comment:
I ask because I believe the `url` shim is brought in either way since browserify doesn't recognize conditional (or dead code) `require`s, so might as well use it